### PR TITLE
Improvements to modpack UX

### LINKED ISSUE #401: Add search box for FTB Legacy
### Role

Modded minecraft experience

### Suggestion

FTB tab does have a search bar, but FTB Legacy doesn't. Why not add it to both?

### Benefit

Common sense :trollface: 

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### LINKED ISSUE #843: [Suggestion] Add Searching for modpacks by project ID
### Role

I happen to be trying out a lot of modpacks that don't have a lot of downloads and are not listed when searching for their name

### Suggestion

Allow using the curseforge project id to be searched for using a prefix ( id:XXXXXX ) where you would ususally just type the name

### Benefit

It is possible, to install modpacks by URL or Zip but causes you to miss out on auto-updates and the curseforge icon. Being able to search by URL also makes sharing packs with friends or community members easier.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### LINKED ISSUE #1239: Consistent UX between modpack providers
### Role

i use prism

### Suggestion

Right now, every modpack provider UI has pretty much a different UX. The only ones that are somewhat consistent are Modrinth and CurseForge. Hell, FTB Legacy doesn't even have a search bar. So my suggestion is, making them consistent and similar to the resource downloader: a search bar with the same text and with a search button (for some reason atlauncher doesn't have it) (perhaps we could also add the instant searching to every modpack page like the resource downloader), and the modpacks have the name and below it their short description if available, while the whole body when clicked, like modrinth and cf right now. Also preserving the search term between the providers would be a nice UX thing

### Benefit

better UX

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### PR DESCRIPTION
**Pull request by Trial97**
fixes #1239
fixes #401
fixes #843

Done
- [x] search bar and button available on all providers
- [x] instant search for all providers(flame,modrinth,technic) 
- [x] modpack name and below description present for all providers (technic doesn't like descriptions)
- [x] you prefix your search term with `#` then followed by the project id will allow you to view the mentioned project(as this is no filter if the project is a modpack or a mod will have the potential danger to display mods in modpack import and also other resources)

I'm not gonna implement the preserving of the search text over different providers.

### CODE REVIEW

**Comment by DioEgizio** [comment]
wouldn't it make more sense to have the loading bar for the instant search like there is for mod downloading? other than that this pr is fine

**Comment by Trial97** [comment]
Just a note regarding the loading bar: some of the searches are done on downloaded information and are instant for those I will not add a loading bar (more exactly only flame, modrinth and technic will have progress bar)

**Review by DioEgizio — APPROVED** [review]

File: launcher/ui/pages/modplatform/ResourceModel.cpp [hunk 1]
```diff
@@ -132,6 +132,36 @@ void ResourceModel::search()
...
+                        return;
+                    searchRequestAborted();
+                };
+
+            if (!callbacks.on_succeed)
+                callbacks.on_succeed = [this](auto& doc, auto& pack) {
+                    if (!s_running_models.constFind(this).value())
+                        return;
+                    searchRequestForOneSucceeded(doc);
+                };
```
Comment (TheKodeToad) [inline launcher/ui/pages/modplatform/ResourceModel.cpp]: why
the bodies of these if statements will always run, right?

Comment (Trial97) [reply launcher/ui/pages/modplatform/ResourceModel.cpp]: yeah you are right(copied from other place where the arguments are not guaranteed to be empty)
fixed now

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**